.github/workflows/build-test-publish-images.yml

```diff
@@ -0,0 +1,254 @@
+name: build and publish imgs workflow
+
+on:
+  workflow_call:
+    inputs:
+      build_type:
+        required: true
+        type: string
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  actions: read
+  checks: none
+  contents: read
+  deployments: none
+  discussions: none
+  id-token: write
+  issues: none
+  packages: read
+  pages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+jobs:
+  compute-matrix:
+    runs-on: ubuntu-latest
+    container:
+      image: rapidsai/ci:latest
+    outputs:
+      BASE_IMAGE_REPO: ${{ steps.compute-image-repo.outputs.BASE_IMAGE_REPO }}
+      NOTEBOOKS_IMAGE_REPO: ${{ steps.compute-image-repo.outputs.NOTEBOOKS_IMAGE_REPO }}
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+      TEST_MATRIX: ${{ steps.compute-test-matrix.outputs.TEST_MATRIX }}
+      BASE_TAG_PREFIX: ${{ steps.compute-tag-prefix.outputs.BASE_TAG_PREFIX }}
+      NOTEBOOKS_TAG_PREFIX: ${{ steps.compute-tag-prefix.outputs.NOTEBOOKS_TAG_PREFIX }}
+      RAPIDS_VER: ${{ steps.compute-rapids-ver.outputs.RAPIDS_VER }}
+      ALPHA_TAG: ${{ steps.compute-rapids-ver.outputs.ALPHA_TAG }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Compute matrix
+        id: compute-matrix
+        run: |
+          MATRIX=$(ci/compute-mx.sh)
```

Comment:
Can we rename `mx` to `matrix`? I think my previous comment got lost. https://github.com/rapidsai/docker/pull/545/#discussion_r1264517848